mos-platform/common/c/atoi.cc

```diff
@@ -0,0 +1,22 @@
+#include <ctype.h>
+#include <stdlib.h>
+
+template <typename T> static inline T _atoi(const char *str) {
+	T result = 0;
+	bool sign = false;
+
+	while (isspace(*str)) { str++; }
+
+	if (*str == '+') { str++; }
+	else if (*str == '-') { sign = true; str++; }
+
+	while (*str >= '0' && *str <= '9') {
+		result = (result * 10) + (*(str++));
```

Comment:
Shouldn't this be `(*(str++) - '0')`?

Reply:
Nice catch!

Made sure to *actually* verify if it works this time...